IETF.js

```diff
@@ -0,0 +1,217 @@
+{
+	"translatorID": "ac4b1d6a-6e52-42e2-9634-ebbe1e96b9ad",
+	"label": "IETF",
+	"creator": "Félix Brezo (@febrezo)",
+	"target": "^https?://tools.ietf.org/(html|pdf|rfc)/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-08-02 02:31:35"
+}
+
+/*
+	IETF Translator
+	Copyright (C) 2020 Félix Brezo (@febrezo), felixbrezo@disroot.org
+	
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	
+	You should have received a copy of the Affero GNU General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+function detectWeb(doc, url) {
+	return "report";
+}
+
+function doWeb(doc, url) {
+	var resourceType = detectWeb(doc, url);
+	var newItem = new Zotero.Item(resourceType);
+	
+	var targetUri;
+	if (url.indexOf("/rfc/") > -1) {
+		targetUri = url.replace("/rfc/", "/html/").replace(".txt", "");
+	}
+	else if (url.indexOf("/pdf/") > -1) {
+		targetUri = url.replace("/pdf/", "/html/").replace(".pdf", "");
+	}
+	else {
+		targetUri = url;
+	}
+
+	Zotero.Utilities.HTTP.doGet(targetUri, function (resText) {
+		// Clean and parse metadata web
+		resText = resText.replace(/<!DOCTYPE[^>]*>/, "").replace(/<\?xml[^>]*\?>/, "");
+		resText = Zotero.Utilities.trim(resText);
+
+		var parser = new DOMParser();
+		var metadataDoc = parser.parseFromString(resText, "text/html");
+
+		// Start scraping
+		newItem.title = ZU.xpathText(metadataDoc, "//meta[@name='DC.Title']/@content");
+		
+		// Iterating through authors
+		var index = 0;
+		while (true) {
+			index++;
+			var tmpAuthor = ZU.xpathText(metadataDoc, "(//meta[@name='DC.Creator'])[" + index + "]/@content");
+			if (tmpAuthor) {
+				var splittedAuthor = tmpAuthor.split(" <")[0].split(", ");	// Remove references to emails that sometimes appear
+				if (splittedAuthor.length == 1) {			// Process authors given as full name
+			 		newItem.creators.push(ZU.cleanAuthor(splittedAuthor[0], "author", false)); 
+				}
+				else {										// Process splitted authors
+					newItem.creators.push({ lastName: splittedAuthor[0], firstName: splittedAuthor[1], creatorType: "author" });
+				}
+			}
+			else {
+				break;
+			}
+		}
+		newItem.reportNumber = "RFC " + ZU.xpathText(metadataDoc, "//meta[@name='DC.Identifier']/@content").split(":")[3];
+		newItem.institution = "IETF";
+		var abstractContent = ZU.xpathText(metadataDoc, "//meta[@name='DC.Description.Abstract']/@content").split("[");
```

Comment:
Same as my comment in the Mastodon PR — a regexp is just going to be much more reliable, since you can anchor the `[FOO]` to the end. It can still be optional, in case it's ever omitted, but you don't want this to break if there happens to be a `[` in the middle of the description.